### PR TITLE
Template the IRCd configuration

### PR DESCRIPTION
**Pull request by KellerFuchs**
### Template the IRC configuration

This PR adds a new playbook, `irc.yml`, which generates the configuration on the IRCds and rehashes as needed.

To support this, the following was done:
- [x] variable files were reorganised to [better support vault](http://docs.ansible.com/ansible/playbooks_best_practices.html#best-practices-for-variables-and-vaults);
- [x] the vault decryption key is kept in `vault_passphrase.gpg`, and decrypted by `vault_passphrase.sh`, see security discussion below;
- [x] the configuration is kept under `files/unrealircd`, as a mixture of plain files and Jinja2 templates.
### Security of Ansible Vault

Ansible vault [does not currently support GnuPG directly](https://github.com/ansible/ansible/pull/7174), so we keep a GPG-encrypted vault decryption key in the repository.

Hence, the vault decryption key is exposed during operation, negating the advantages of smart cards. However:
- the data being decrypted is seldom changed;
- the decryption key is versioned along the data;
- the decryption key is never used for other data.

As such, it is possible to change the decryption key every time the data is changed, completely solving this issue.  It's however inconvenient to enforce, but hints that regular vault key rollovers would be an effective mitigation.

**TL;DR**: Replacing the vault key from time to time is fine.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
Tentatively assigning @ChickenNuggers (since he is an IRC guru) and @drGrove (since [he was interested](https://github.com/hashbang/admin-tools/pull/2#discussion_r52422274)).

File: files/unrealircd/opers.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# {{ ansible_managed }}
+
+### Admins
+{% for user in user_groups.admins %}
+
+oper {{ user }} {
+	class clients;
+	from { userhost *@*; };
+	password "{{ users[item].irc_oper.pass }}" { {{ users[item].irc_oper.type }}; };
```
Comment (RyanSquared) [inline files/unrealircd/opers.conf.j2]: Should `users[item]` be `users[user]` because of line 4?

File: files/unrealircd/opers.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{% endfor %}
+
+
+# IRCops
+{% for user in user_groups.irc_ops %}
+
+oper {{ user }} {
+        class clients;
+        from { userhost *@*; };
+	password "{{ users[item].irc_oper.pass }}" { {{ users[item].irc_oper.type }}; };
```
Comment (RyanSquared) [inline files/unrealircd/opers.conf.j2]: Should `users[item]` be `users[user]` because of line 4?

**Comment by KellerFuchs** [comment]
@ChickenNuggers Good catch! (And fixed)

**Comment by KellerFuchs** [comment]
@hashbang/administrators @hashbang/contributers If somebody else was to review this, I would feel much more confident about trying it out.
(Not that I don't expect stuff to fail initially, but at least it ought to fail in a non user-visible way)

**Comment by drGrove** [comment]
@KellerFuchs, can you make the alias files a little more DRY?

**Comment by KellerFuchs** [comment]
@drGrove Done
